src/Schema/AbstractIntrospectingSchemaManager.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Doctrine\DBAL\Schema;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Exception\DatabaseRequired;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Result;
+
+use function array_shift;
+
+/**
+ * Base class for schema managers that improves database schema introspection performance.
```

Comment:
Speaking of improvements makes sense in a commit message or PR, but in code, one might wonder what is the slow version of this.

```suggestion
 * Base class for schema managers that provides good database schema introspection performance.
```

Reply:
Maybe, just avoid qualitative statements at all

>  Base class for schema managers that provides database schema introspection.

Reply:
IMO, the original "improves" is more appropriate than "provides good". But I believe I want to get rid of this class entirely as follows:

1. Declare the new abstract methods in the existing class as non-abstract with the `throw new NotImplemented()` body.
2. Create temporary deprecated methods that leverage the new API (e.g. `protected doListTableDetails()`).
3. In those classes that want to opt into the new API (all bundled schema managers), implement the abstract methods (already done) and instead of inheriting the default implementation of `listTableDetails()` from the base class, call `doListTableDetails()`.

Otherwise, if we want to do more rework like this in the next minor release (e.g. listing sequences, users, etc.), we'll have to introduce another temporary abstract class turning it into a mess.